packages/webview_flutter/webview_flutter/ios/Classes/FlutterWebView.m

```diff
@@ -230,18 +238,31 @@ - (void)onEvaluateJavaScript:(FlutterMethodCall*)call result:(FlutterResult)resu
                                details:nil]);
     return;
   }
-  [_webView evaluateJavaScript:jsString
-             completionHandler:^(_Nullable id evaluateResult, NSError* _Nullable error) {
-               if (error) {
-                 result([FlutterError
-                     errorWithCode:@"evaluateJavaScript_failed"
-                           message:@"Failed evaluating JavaScript"
-                           details:[NSString stringWithFormat:@"JavaScript string was: '%@'\n%@",
-                                                              jsString, error]]);
-               } else {
-                 result([NSString stringWithFormat:@"%@", evaluateResult]);
-               }
-             }];
+  void (^evalResultHandler)(id, NSError*) = ^(_Nullable id evaluateResult,
+                                              NSError* _Nullable error) {
+    if (error) {
+      result([FlutterError
+          errorWithCode:@"evaluateJavaScript_failed"
+                message:@"Failed evaluating JavaScript"
+                details:[NSString
+                            stringWithFormat:@"JavaScript string was: '%@'\n%@", jsString, error]]);
+    } else if (evaluateResult == /*null*/ [NSNull null] || evaluateResult == /*undefined*/ nil) {
+      result(@"null");
+    } else {
+      result([NSString stringWithFormat:@"%@", evaluateResult]);
+    }
+  };
+  if (@available(iOS 14, *)) {
+    jsString = [NSString
+        stringWithFormat:@"return eval(\"%@\");", [self escapeStringForJavaScript:jsString]];
```

Comment:
What's the reason for wrapping this in an `eval`? That seems like something that could cause side-effects. E.g., I believe CSP could completely break this.

Reply:
The reason for this is that unlike `evaluateJavascript`, `callAsyncJavaScript` executes the given string as the body of an async function, rather than directly evaluating it. As such, it will only return any value that is explicitly returned, not the result of the string being evaluated. I use `eval` here to make sure the behavior is identical to `evaluateJavascript` on iOS 13 and below. 

Personally I have not been able to find any edge cases or side effects, but I've also not tried anything to do with CSP. If you have any examples of this maybe we can find a way around it?

The only other way I can think of to avoid these side effects, assuming they're a thing, is to fix this the other way around:(make `evaluateJavaScript` behave like `callAsyncJavaScript`), forcing everyone to explicitly return a value in their JS strings, and consider it a breaking change.

Please let me know what you think. In the mean time I'll update this PR to move the fix to the separated iOS implementation.

Reply:
> (make `evaluateJavaScript` behave like `callAsyncJavaScript`), forcing everyone to explicitly return a value in their JS strings, and consider it a breaking change.

How would that be better than the current behavior?

Reply:
I mean master; isn't the issue you are trying to fix that a void return causes an error? Can't people already avoid that error right now by explicitly returning a value?

Reply:
On master the platform exception can indeed be avoided by ensuring the given string evaluates to a non-null/non-undefined value.

E.g:
`Snackbar.postMessage("User Agent: " + navigator.userAgent)` Throws.
`Snackbar.postMessage("User Agent: " + navigator.userAgent);1` Does not.